components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,538 @@
+		}
+	end
+end
+
+-- called from Module:Match/Subobjects
+function CustomMatchGroupInput.processPlayer(player)
+	return player
+end
+
+--
+--
+-- function to sort out winner/placements
+function CustomMatchGroupInput._placementSortFunction(table, key1, key2)
+	local op1 = table[key1]
+	local op2 = table[key2]
+	local op1norm = op1.status == _STATUS_HAS_SCORE
+	local op2norm = op2.status == _STATUS_HAS_SCORE
+	if op1norm then
+		if op2norm then
+			local op1setwins = CustomMatchGroupInput._getSetWins(op1)
+			local op2setwins = CustomMatchGroupInput._getSetWins(op2)
+			if op1setwins + op2setwins > 0 then
+				return op1setwins > op2setwins
+			else
+				return tonumber(op1.score) > tonumber(op2.score)
+			end
+		else return true end
+	else
+		if op2norm then return false
+		elseif op1.status == _STATUS_DEFAULT_WIN then return true
+		elseif Table.includes(ALLOWED_STATUSES, op1.status) then return false
+		elseif op2.status == _STATUS_DEFAULT_WIN then return false
+		elseif Table.includes(ALLOWED_STATUSES, op2.status) then return true
+		else return true end
+	end
+end
+
+function CustomMatchGroupInput._getSetWins(opp)
+	local extradata = opp.extradata or {}
+	local set1win = extradata.set1win and 1 or 0
+	local set2win = extradata.set2win and 1 or 0
+	local set3win = extradata.set3win and 1 or 0
+	local sum = set1win + set2win + set3win
+	return sum
+end
+
+--
+-- match related functions
+--
+function matchFunctions.readDate(matchArgs)
+	return matchArgs.date
+		and MatchGroupInput.readDate(matchArgs.date)
+		or {
+			date = MatchGroupInput.getInexactDate(globalVars:get('tournament_enddate')),
+			dateexact = false,
+		}
+end
+
+function matchFunctions.getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', '2v2'))
+	match.showh2h = Logic.emptyOr(match.showh2h, Variables.varDefault('showh2h'))
+	return MatchGroupInput.getCommonTournamentVars(match)
+end
+
+function matchFunctions.getVodStuff(match)
+	match.stream = Streams.processStreams(match)
+	match.vod = Logic.emptyOr(match.vod, Variables.varDefault('vod'))
+
+	-- apply vodgames
+	for index = 1, MAX_NUM_VODGAMES do
+		local vodgame = match['vodgame' .. index]
+		if not Logic.isEmpty(vodgame) then
+			local map = match['map' .. index] or {}
+			map.vod = map.vod or vodgame
+			match['map' .. index] = map
+		end
+	end
+	return match
+end
+
+function matchFunctions.getExtraData(match)
+	local opponent1 = match.opponent1 or {}
+	local opponent2 = match.opponent2 or {}
+
+	local casters = {}
+	for key, name in Table.iter.pairsByPrefix(match, 'caster') do
+		table.insert(casters, CustomMatchGroupInput._getCasterInformation(
+			name,
+			match[key .. 'flag'],
+			match[key .. 'name']
+		))
+	end
+	table.sort(casters, function(c1, c2) return c1.displayName:lower() < c2.displayName:lower() end)
+
+	local showh2h = Logic.readBool(match.showh2h)
+		and opponent1.type == Opponent.team
+		and opponent2.type == Opponent.team
+
+	match.extradata = {
```

Comment:
is any of the below stuff that gets added to extradata (except casters) needed?

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2259/commits/21d2fd191cfa4610aac0e37e794a927855713c62